src/Server.cpp

```diff
@@ -156,43 +153,37 @@ void Server::handleConnection(Connection conn, size_t &i)
 	{
 		std::cout << "Path exists" << std::endl;
 
-		if (router.isDynamicRequest(request))
+		// if (router.isDynamicRequest(request))
+		// {
+		// if (request.getMethod() == "GET" && request.getRequestTarget() == "/hello.cgi")
+		if (request.getMethod() == "GET")
 		{
-			if (request.getMethod() == "GET" && request.getRequestTarget() == "/hello")
-			{
-				// env has to be created before CGI, because it is passed to the CGI
-				CGIHandler cgiHandler;
-				Environment env;
-				env.setVar("QUERY_STRING", "Hello from C++ CGI!");
-				// cgiHandler.executeCGI(argv, env);
-				handleCGIRequest(argv, env);
-			}
-			else
-			{
-				CGIHandler cgiHandler;
-				Environment env;
-				env.setVar("request.getQueryString()", "request.getBody()");
-				// response = cgiHandler.handleCGIRequest(argv, request);
-				// cgiHandler.executeCGI(argv, env);
-				handleCGIRequest(argv, env);
-			}
+			CGIHandler cgiInstance;
+			Environment env;
+			env.HTTPRequestToMetaVars(request, env);
+			response = cgiInstance.handleRequest(request);
 		}
 		else
 		{
-			StaticContentHandler staticContentHandler;
-			// This if condition only for legacy reasons! TODO: remove
-			if (request.getMethod() == "GET" &&
-				(request.getRequestTarget() == "/" || request.getRequestTarget() == "/home"))
-			{
-				response = staticContentHandler.handleHomePage();
-			}
-			else
-			{
-				response = staticContentHandler.handleRequest(request);
-			}
+			std::cout << "Dynamic request not supported" << std::endl;
 		}
+		// }
+		// else
+		// {
+		// 	StaticContentHandler staticContentHandler;
+		// 	// This if condition only for legacy reasons! TODO: remove
+		// 	if (request.getMethod() == "GET" &&
+		// 		(request.getRequestTarget() == "/" || request.getRequestTarget() == "/home"))
+		// 	{
+		// 		response = staticContentHandler.handleHomePage();
+		// 	}
+		// 	else
+		// 	{
+		// 		response = staticContentHandler.handleRequest(request);
+		// 	}
+		// }
 	}
-	std::string responseString = response.toString();
+	std::string responseString = response.getBody();
 
```

Comment:
This is conceptually "wrong" and the previous version was in my opinion right. But I could be wrong. We want to write the whole response into the socket and not the body. So if it works with getBody probably get the whole response from response object. Can you explain me this? Cause I'm confused.

Reply:
toString transforms a response object into a string, we used it for "normal" responses
getBody was used for CGI responses where the whole cgi output was strored in the body...

The second way was not elegant so now (Pr is on its way) we get the CGI output and separate it into header and body. So Router always receive a response, and we use only `toString`

Reply:
Ok. So this will just change.